lib/app_web/controllers/api_controller.ex

```diff
@@ -0,0 +1,49 @@
+defmodule AppWeb.ApiController do
+  use AppWeb, :controller
+  import SweetXml
+
+  def create(conn, %{"image" => image}) do
+
+    # Check if file is an image
```

Comment:
feels like we're duplicating code in this controller that could be in a shared library? 💭

Reply:
I don't understand. Where is this code being duplicated?

Reply:
upload code abstracted to: https://github.com/dwyl/imgup/blob/40e95f493c926f21cb5877ed456fba01cc9e13ca/lib/app/upload.ex
and end-to-end tested: 
https://github.com/dwyl/imgup/blob/40e95f493c926f21cb5877ed456fba01cc9e13ca/test/app/upload_test.exs